tests/TestCase/Http/Client/RequestTest.php

```diff
@@ -37,7 +48,44 @@ public function testConstructorStringData()
         $request = new Request('http://example.com', 'POST', $headers, json_encode($data));
 
         $this->assertEquals('http://example.com', $request->url());
-        $this->assertEquals('POST', $request->getMethod());
+        $this->assertContains($request->getMethod(), 'POST');
+        $this->assertEquals('application/json', $request->getHeaderLine('Content-Type'));
+        $this->assertEquals(json_encode($data), $request->body());
+    }
+
+    /**
+     * test string ata, header and constructor
+     *
+     * @return void
+     */
+    public function testAllRequestMethods()
+    {
+        $headers = [
+            'Content-Type' => 'application/json',
+            'Authorization' => 'Bearer valid-token',
+        ];
+        $data = ['a' => 'b', 'c' => 'd'];
+        $request = new Request('http://example.com', 'POST', $headers, json_encode($data));
+        $this->assertContains($request->getMethod(), $this->methods);
+        $this->assertEquals('http://example.com', $request->url());
+        $this->assertEquals('application/json', $request->getHeaderLine('Content-Type'));
+        $this->assertEquals(json_encode($data), $request->body());
```

Comment:
This looks like a test that could benefit from a [dataprovider](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers).